README.md

```diff
@@ -1,42 +1,89 @@
+
+All DSAD APIs return JSON responses and, where applicable, accept JSON request bodies. No other serialization formats are currently supported.
+
+Except where otherwise noted, DSAD APIs comply with CareerBuilder API standards with respect to request and response syntax. Specifically, all multi-word request and response field names are written in `lower_snake_case` and all successful API responses are returned as a JSON object with a single top-level "data" object field containing all other response data. Error responses are returned as a JSON object with a single top-level "errors" array field containing one or more objects, defined as follows: `{ type: String, message: String, code: int }`.
+
+### Common Error Codes
+
+Following is a non-exhaustive list of error codes that may be returned by DSAD APIs. Most of our APIs can return most of these codes in the cases described here. Additional codes, or deviations from the behavior-to-code mappings defined here, may be described in API-specific documentation pages.
+
+Note that the CareerBuilder API routing layer or OAuth token endpoint may also occasionally return HTTP errors. In particular, any non-standard error code in the "high 500s" (such as HTTP 572) is generated by the routing layer, not by a DSAD API. Consult the Authorization team for information on these error codes.
+
+* **400 (Bad Request)** : The API request was missing a required field or was otherwise invalid.
+* **401 (Unauthorized)** : The API request was not authorized. The API may only accept requests from whitelisted client_ids, or the request may not have been transmitted through the CB API routing layer properly.
+* **404 (Not Found)** : The API request targeted a route (or a version of a route) that does not exist.
+* **405 (Method Not Allowed)** : The API request specified an HTTP method that is not supported by the targeted API route and version.
+* **429 (Too Many Requests)** : The client_id associated with this request has sent too many requests to this API recently, and the API's performance is beginning to degrade as a result. Please wait a few minutes before resuming API calls.
+* **500 (Internal Server Error)** : The API request was accepted, but failed to complete successfully due to an internal error within the API. This error should not occur under normal circumstances; if you encounter this, please contact the DSAD team.
+* **502 (Bad Gateway)** : The API request was accepted, but failed to complete successfully due to an error which occurred with a downstream provider called during request processing. This error should not occur under normal circumstances; if you encounter this, please contact the DSAD team.
```

Comment:
I think it would be an upstream provider here. Also we don't want people calling us on any one 500, would it be better to state that we have a threshold of acceptable 500 depending the service and if you see consistent errors then contact us?

Reply:
@msundlof yeah, you're right, upstream is correct here. I'll fix that.

I don't think we should generally accept any steady-state percentage of 500s. to me, any 500 from our API indicates a mistake on our part to be fixed -- either something that *should* be a 4xx, or a programming error that shouldn't happen at all. (note I'm referring to 500 specifically, not all 5xx's, though I also feel that we should work toward 0% steady-state 5xx's across our platform as well).

Reply:
so I'd like to keep the "please contact us" verbiage for 500s, but I'm open to changing it for 502s. I'll bring it up in the team room and see what others think.